### PR TITLE
Overhauled output of converting to E2 holograms

### PR DESCRIPTION
**Pull request by DatAmazingCheese**
Overhauled it to make it less messy, and require 0 cleanup. 

Went from this:

To this:

### CODE REVIEW

File: lua/pac3/editor/client/tools.lua [hunk 1]
```diff
@@ -276,84 +276,76 @@ do
...
-		holoAlpha(HOLO_NAME, ALPHA)
-		holoScale(HOLO_NAME, SCALE)
-		holoPos(HOLO_NAME, entity():toWorld(POSITION))
-		holoAng(HOLO_NAME, entity():toWorld(ANGLES))
-		#holoAnim(HOLO_NAME, ANIMATION_NAME, ANIMATION_FRAME, ANIMATION_RATE)
-		#holoDisableShading(HOLO_NAME, FULLBRIGHT)
-		holoMaterial(HOLO_NAME, MATERIAL)
-		holoModel(HOLO_NAME, MODEL)
-		holoSkin(HOLO_NAME, SKIN)
+		holoColor(I, COLOR)
```
Comment (DatAmazingCheese) [inline lua/pac3/editor/client/tools.lua]: Cleaned up how the holograms are laid out, now auto indexes and doesn't include unimplemented stuff.

File: lua/pac3/editor/client/tools.lua [hunk 2]
```diff
@@ -276,84 +276,76 @@ do
...
 		end
-
-		local holo = holo_str
-		:gsub("IDX", part.UniqueID)
+		
+		local holo = holo_str			
 		:gsub("ALPHA", part:GetAlpha()*255)
 		:gsub("COLOR", tovec(part:GetColor()))
-		:gsub("SCALE", tovec(Vector(scale.y, scale.x, scale.z)))
+		:gsub("SCALE", tovec(Vector(scale.x, scale.y, scale.z)))
```
Comment (DatAmazingCheese) [inline lua/pac3/editor/client/tools.lua]: Fixed scale, which was a huge problem when cleaning up.

File: lua/pac3/editor/client/tools.lua [hunk 3]
```diff
@@ -276,84 +276,76 @@ do
...
 		else
-			holo = holo:gsub("PARENT", "holoParent(HOLO_NAME, entity())")
+			holo = holo:gsub("PARENT", "holoParent(I, entity())")
 		end
-
-		holo = holo:Replace("HOLO_NAME", "PAC_" ..part:GetName():gsub("%p", ""):gsub(" ", "_"))
-
-		LocalPlayer():ChatPrint("PAC -> E2 holo code printed to console.")
+		
+		holo = holo:Replace("HOLO_NAME", part:GetName())
```
Comment (DatAmazingCheese) [inline lua/pac3/editor/client/tools.lua]: No more PAC_ crap. Keeping it simple

**Comment by LUModder** [comment]
Is parenting fixed cause that was a big issue converting was patented parts

**Comment by LUModder** [comment]
Also convering from a group as the root wouldn't make it do anything

**Comment by DatAmazingCheese** [comment]
I've never heard of parenting being an issue. I don't parent to other holos, I just parent to the 1 entity. Then in code I parent stuff to dummies etc.

**Comment by CapsAdmin** [comment]
Can you remove the extra whitespace?

**Comment by DatAmazingCheese** [comment]
Not 100% sure what you mean, if you mean the extra spaces dotted around I didn't touch them. First real time using Github, I just really wanted to fix the horrid-ness of how it outputs the holograms. I use it a lot when I build and its a pain to clean up.

**Pull request closed without merging**